### ISSUE TITLE
Push Notification provider choice (UnifiedPush)

### ISSUE THREAD

**Issue by alien999999999** [BelledonneCommunications/linphone-android#1514]
It would be great to have a checkbox under allow Push Notifications, that would use UnifiedPush (you should probably mention that you need to install a specific app though)

**Comment by Viish** [comment]
I'll leave the issue open with the "feature request" tag but clearly this won't be done soon.

**Comment by alien999999999** [comment]
i understand, but this could relieve a lot of support questions on push notifications, as you wouldn't have to go through the google servers, with its api key requirements for the pusher

**Comment by 143981** [comment]
Please do this, for the f-droid version it is very necessary... linphone is the latest application using google apps for me

**Comment by alien999999999** [comment]
Any progress on this? this issue does seem to have a few people who want this, and independance from google would be great, but atm, I switched to sipnetic

**Comment by Viish** [comment]
> Any progress on this?

No, this isn't on our roadmap for the time being.

**Comment by lnpn23121** [comment]
Google is blocked in some countries, without this feature, push notification can't be used, so it's not just for people who choose to avoid using Google, it's also for people who can't use Google. You could use ntfy for testing so you don't need to set up the push notification server.

**Comment by anaxonda** [comment]
> @lnpn23121 We have a "background mode" for Android that can be used when push are unavailable that works quite well.

Background mode has quite high battery use, right?

**Comment by Viish** [comment]
@anaxonda obviously the battery cost is higher than not having the app alive at all times but it's not that high, we have made multiple changes to keep it to a minimum.

**Comment by alien999999999** [comment]
the biggest problem with that is that there's a bunch of phones that just destroy background apps :-(

**Comment by EmanuelLoos** [comment]
Please consider issue #2171. Background mode is broken on many devices including my device. Implementing UnifiedPush would fix this:

https://unifiedpush.org/
https://f-droid.org/de/2026/01/08/unifiedpush-5-years.html

**Comment by EmanuelLoos** [comment]
With UnifiedPush users can flexibly select one of many apps with working background services including the [very stable XMPP client Conversations](https://unifiedpush.org/users/distributors/conversations/), simple apps like [Sunup](https://unifiedpush.org/users/distributors/sunup/) or [ntfy](https://unifiedpush.org/users/distributors/ntfy/) that work without an account, [NextPush which integrates with Nextcloud](https://unifiedpush.org/users/distributors/nextpush/) or even the [gCompat UP-Distributor that uses the Google Play Services](https://unifiedpush.org/users/distributors/fcm/) to be their distributor for Push notifications. That way only a single battery-draining background service is needed to get Push notifications for a multitude of apps. [Many apps use UnifiedPush already.](https://unifiedpush.org/users/apps/)

It can be implemented using the following Android libraries:
- https://unifiedpush.org/kdoc/connector/
- https://unifiedpush.org/kdoc/connector_ui/

And using WebPush server-side.

See here for a simple introduction: https://unifiedpush.org/developers/intro/

**Comment by tctlrd** [comment]
Agreed! Linphone is great; it will greatly benefit from the integration of [UnifiedPush](/UnifiedPush).

**Comment by distinctjuggle** [comment]
This is the most requested (by thumbs up reactions) issue for the entire project and it's not even close.  One would think there would be more of a care by the development team to integrate this in the past nearly 5 years since it was originally requested.  

UnifiedPush is the standard for FOSS push notifications, and the community would very surely appreciate this addition as evidenced by this thread's activity.  Personally I just found this app and was excited to start using it until I saw the apathy demonstrated here.  It's discouraging to know that if I use this my phone will have a shorter lifespan due to a lack of concern from the developers, on what is a 100% solved problem.  I really hope this could see some attention!

**Comment by basncy** [comment]
Before officially support this feature, some of you could try running ntfy via shizuku in the background, subscribe to a device-specific topic, and trigger am start $ACTIVITY command to wake up linphone.
On the flexisip/asterisk server side, configure [external-push-uri](https://wiki.linphone.org/xwiki/wiki/public/view/Flexisip/Core%20servers/Configuration%20reference%20guide/2.3.3/module/PushNotification) to send an HTTP request to wake Linphone Android up.

**Comment by tctlrd** [comment]
@basncy could you explain what you mean by running ntfy via shizuku in the background? As far as I know ntfy is not yet setup to utilize the shizuku API.

**Comment by schklom** [comment]
@tctlrd You could use an automation app that can read and delete notifications and integrate with ADB via shizuku, e.g. Tasker.

notification via ntfy (maybe broadcast?) -> Tasker notices it -> Tasker deletes the notification and triggers the ADB command

EDIT: Tasker can react to NTFY receiving a message, it doesn't need to read the notification itself. So this can be done completely silently if you also disable showing the notification category on ntfy. Example on https://docs.ntfy.sh/subscribe/phone/#react-to-incoming-messages

**Comment by tctlrd** [comment]
@schklom yes that is a solution.

**Comment by basncy** [comment]
First, you have to make NTFY-aarch64-musl  background running. Second, the caller of "am start $ACTIVITY" should have the correct permission.
That's why I suggest Shizuku as the environment for non-rooted devices.  Introducing Tasker or MacroDroid is yet another approach, but you have to keep ntfy.apk running in the background.
If you have a rooted device, running ntfy with root should be a perfect way.

https://docs.ntfy.sh/subscribe/cli/#subscribe-to-multiple-topics

**Comment by sk425** [comment]
UnifiedPush support would be a very useful feature to me.

The background method works until memory pressure increases beyond capacity for a foreground app, then android will silently kill background apps to make room. I don't know its killed until I check all my  background app keep-alive notifications (linphone isn't the only one), or I open the app to call someone. Between then, I'm missing calls. 

Background usage also increases battery usage as mentioned above, but I'm happy this has been improved over the years

Shizuku method I'm sure could work but I don't turn on developer settings as it increases security risk.
